cloud/aws/cli/core.py

```diff
@@ -141,16 +148,37 @@ def current_lambda_image(c, repo_arn):
     return current
 
 
-@task()
+@task
 def deploy_lambda_image(c):
     """Build and push the lambda image, fails if step 1 is not deployed"""
-    image_url = terraform_output(c, "step-1", "vast_lambda_repository_url")
+    image_url = terraform_output(c, "core-1", "vast_lambda_repository_url")
+    repo_arn = terraform_output(c, "core-1", "vast_lambda_repository_arn")
+    # get the digest of the current image
+    current_image = current_lambda_image(c, repo_arn)
+    try:
+        c.run(f"docker pull {current_image}")
+        old_digest = c.run(
+            f"docker inspect --format='{{{{.RepoDigests}}}}' {current_image}",
+            hide="out",
+        ).stdout
+    except:
+        old_digest = "current-image-not-found"
+    # build the image and get the new digest
     image_tag = int(time.time())
     c.run(
-        f"docker build --build-arg VAST_VERSION={VAST_VERSION(c)} -f docker/lambda.Dockerfile -t {image_url}:{image_tag} ./docker"
+        f"docker build --build-arg VAST_VERSION={VAST_VERSION(c)} -f {DOCKERDIR}/lambda.Dockerfile -t {image_url}:{image_tag} {DOCKERDIR}"
     )
+    new_digest = c.run(
+        f"docker inspect --format='{{{{.RepoDigests}}}}' {image_url}:{image_tag}",
+        hide="out",
+    ).stdout
+    # compare old an new digests
+    if old_digest == new_digest:
+        print("Docker image didn't change, skipping push")
+        return
```

Comment:
docker actually does this check internally to avoid pushing layers that didn't change but here we do it explicitly to avoid creating a new tag in ECR and moving the `current` pointer to it. This spares us a systematic state change of the Lambda when running the next Terraform step